packages/jaeger-ui-components/src/scroll-page.test.ts

```diff
@@ -19,17 +19,18 @@ import { scrollBy, scrollTo, cancel } from './scroll-page';
 
 // keep track of instances, manually
 // https://github.com/facebook/jest/issues/5019
-const tweenInstances = [];
+const tweenInstances: Tween[] = [];
 
 describe('scroll-by', () => {
   beforeEach(() => {
     window.scrollY = 100;
     tweenInstances.length = 0;
-    Tween.mockClear();
-    Tween.mockImplementation((opts) => {
-      const rv = { to: opts.to, onUpdate: opts.onUpdate };
+    jest.mocked(Tween).mockClear();
+    jest.mocked(Tween).mockImplementation((opts) => {
+      const rv = { to: opts.to, onUpdate: opts.onUpdate } as Tween;
       Object.keys(Tween.prototype).forEach((name) => {
         if (name !== 'constructor') {
+          // @ts-ignore
           rv[name] = jest.fn();
         }
       });
```

Comment:
The last time we added a //@ts-ignore it caused a bug!

Best not to add this unless completely necessary. Is this Object.keys even necessary?

Reply:
This loop was adding `onUpdate` and `cancel` properties to the `rv` object and setting their values to `jest.fn()`. I have now removed this loop (so `//@ts-ignore` is no longer needed) and manually added `onUpdate` and `update`. See commit https://github.com/grafana/grafana/pull/59617/commits/9fd536bfda6cdf1125f0688c652d7a322c5f7d52.